src/smpp_socket.erl

```diff
@@ -471,10 +475,15 @@ handle_pkt(#pdu{command_id = CmdID, sequence_number = Seq} = Pkt,
                             rate_limit_cooldown -> StateName;
                             _ -> bound
                         end,
-            {ok, NextState, State1};
+            State2 = case State1 of
+                #{in_flight := []} -> maps:without([response_time], State1);
+                #{in_flight := _} -> set_request_timeout(State1)
```

Comment:
I'd move this to additional clause of `set_request_timeout/1` as the latter one incapsulates the handling of `response_time` state variable. E.g
```erlang
set_request_timeout(#{in_flight := []} = State) ->
    aps:without([response_time], State);
set_request_timeout(#{in_flight := InFlight} = State) ->
    ...
```

Reply:
Done